packages/devtools_app/lib/src/screens/app_size/app_size_controller.dart

```diff
@@ -161,21 +175,64 @@ class AppSizeController {
   }
 
   TreemapNode? get _activeDiffRoot {
-    switch (_activeDiffTreeType.value) {
-      case DiffTreeType.increaseOnly:
-        return _increasedDiffTreeRoot;
-      case DiffTreeType.decreaseOnly:
-        return _decreasedDiffTreeRoot;
-      case DiffTreeType.combined:
-      default:
-        return _combinedDiffTreeRoot;
+    final isDeferred = _isDeferredApp.value;
```

Comment:
If we follow the suggestion I made below, we can simplify this getter considerably. We can add another getter for the `_activeDiffMap` which returns either the `_diffMap`, `_mainDiffMap`, or `_deferredDiffMap`. Then, this `getter` only needs to return either `_activeDiffMap.combined`, `_activeDiffMap.increaseOnly` or `_activeDiffMap.decreaseOnly`

Reply:
Looks better, but adding an `_activeDiffMap` getter (as described above) will clean this up even more!